tests/suites/os/tests/connectivity/index.js

```diff
@@ -74,7 +74,25 @@ module.exports = {
 				return {
 					title: `${proxy.charAt(0).toUpperCase()}${proxy.slice(1)} test`,
 					run: async function(test) {
-						const PROXY_PORT = 8123;
+						let getProxyContainerID = async() => {
+							return this.context
+								.get()
+								.worker.executeCommandInHostOS(
+									['balena', 'ps', '-qf', 'name=proxy'].join(' '),
+									this.context.get().link
+								);
+						};
+
+						const ip = await this.context.get()
+							.worker.ip(this.context.get().link);
+
+						// Ensure we only push and run the proxy container once
+						if (!await getProxyContainerID()) {
+							test.comment('Running proxy in container');
+							const state = await this.context.get()
+								.worker.pushContainerToDUT(ip, __dirname, 'proxy');
```

Comment:
Does this line actually verify that the container name is `proxy`? From the source it looks like that doesn't set the name but only verifies it, so it must get it from the compose file? Does it not use a prefix in that case?

https://github.com/balena-os/leviathan/blob/7a44f00b8c71a076e44ae58bd676ac05f700fc21/core/lib/common/worker.js#L259

Asking because if we verify the container name when creating it, we can skip the step of getting the container ID in order to read the logs, yes?

Reply:
@klutchell Exactly, during development, I hit a scenario where the second run of this test pushed the container again, causing there to be two running containers. We only want it running once.